docs/running_psalm/plugins/plugins_type_system.md

```diff
@@ -174,15 +174,15 @@ Note that these classes do sometimes change, so `Type::parseString` is always go
 
 ### Creating type object instances directly
 
-The following example constructs a types representing a string, a floating point number, and a class called 'Foo\Bar\SomeClass'.
+The following example constructs types representing a string, a floating-point number, and a class called 'Foo\Bar\SomeClass'.
 
 ``` php
 new TLiteralString('A text string')
 new TLiteralFloat(3.142)
 new TNamedObject('Foo\Bar\SomeClass')
 ```
 
-Types within Psalm are always wrapped in a union as a convenience feature. Almost anywhere you may expect a type, you can get a union as well (property types, return types, argument types, etc). So wrapping a single atomic type (like TInt) in a union container allows to uniformly handle that type elsewhere, without repetitive checks like this:
+Psalm always wraps types in a union as a convenience feature. Almost anywhere you may expect a type, you can get a union as well (property types, return types, argument types, etc). So wrapping a single atomic type (like TInt) in a union container allows to uniformly handle that type elsewhere, without repetitive checks like this:
```

Comment:
This sounds like it's done automatically, but it's actually not.

Reply:
@weirdan yes thas has been turned in active language. If it's not psalm what is the subject that does the wrapping?

Or is it that this is a concept and while psalm evolves it has turned out this is a good model?

Reply:
In the context of plugins it could be the plugin itself. E.g when you need to provide a return type to Psalm in a method return type provider you would have to construct the atomic type and wrap it into a union, then return it from the hook.

Reply:
Sounds to me as if this is completely optional. I've reverted therefore to keep it passive so any plugin author can figure out their own whether or not the composite pattern is to theirs benefit or they restrain from applying it. Squashed all these changes to reset the PR.

Also removed the change of the error messages as it's too invasive for a simple PR.